contrib/checkbox-dss-validation/checkbox-provider-dss/bin/check_notebook.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
+#
+# This file is part of Checkbox.
+#
+# Copyright 2022 Canonical Ltd.
+#
+# Authors:
+#     Abdullah (@motjuste) <abdullah.abdullah@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""Check notebooks in DSS"""
+
+import subprocess
+import textwrap
+import typing as t
+
+from common import create_parser_with_checks_as_commands
+
+_TIMEOUT_SEC: float = 15.0 * 60  # seconds
+
+SCRIPT = {
+    "pytorch_is_available": textwrap.dedent(
+        """
+        import torch
+        print(torch.__version__)
+        """
+    ),
+    "tensorflow_is_available": textwrap.dedent(
+        """
+        import tensorflow as tf
+        print(tf.config.experimental.list_physical_devices())
+        """
+    ),
+    "pytorch_can_use_intel_gpu": textwrap.dedent(
+        """
+        import sys
+        import torch
+        import intel_extension_for_pytorch as ipex
+
+        print(torch.__version__)
+        print(ipex.__version__)
+
+        if torch.xpu.device_count() < 1:
+            raise AssertionError("no XPUs are available")
+
+        [
+            print(i, torch.xpu.get_device_properties(i))
+            for i in range(torch.xpu.device_count())
+        ]
```

Comment:
Why are you using list comprehension here?
List comprehensions are meant to create lists, not execute side effects like printing.
Just add a for loop here if you have to print the detected xpu:
```suggestion
        for i in range(torch.xpu.device_count()):
            print(i, torch.xpu.get_device_properties(i))
```

Reply:
The script has been updated and this unnecessary part is actually removed.  See the script `data/sees_intel_gpu_in_pytorch.py`.